File: 0283-move-zeroes/0283-move-zeroes.go

func moveZeroes(nums []int) {
	l := 0
	for r := range nums {
		if nums[r] != 0 {
			nums[l] = nums[r]
			l += 1
		}
	}

	for l < len(nums) {
		nums[l] = 0
		l += 1
	}
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        indx = 0           
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[indx], nums[i] = nums[i], nums[indx]
                indx += 1
        
        return
